typescript/sdk/src/contracts.ts

```diff
@@ -1,79 +1,52 @@
-import { Contract } from 'ethers';
+import { ethers } from 'ethers';
 
-import { Router__factory } from '@abacus-network/app';
-import { AbacusConnectionManager__factory } from '@abacus-network/core';
 import { types } from '@abacus-network/utils';
 
-import { Connection, ProxiedAddress } from './types';
+import { Connection } from './types';
+import { objMap } from './utils';
 
-// address types generated from AbacusDeployer deployContract or deployProxiedContract
-export type AbacusContractAddresses = {
-  [key in string]: ProxiedAddress | types.Address;
+export type AbacusFactories = {
+  [key: string]: ethers.ContractFactory;
 };
 
-// from concrete ethers.ContractFactory static connect() type
-type EthersFactory<C> = (
-  address: types.Address, // TODO: generic on ProxiedAddress for proxy utilities
-  connection: Connection,
-) => C;
-export type Factories<A extends AbacusContractAddresses> = Record<
-  keyof A,
-  EthersFactory<any>
->;
+export type AbacusContracts = {
+  [key: Exclude<string, 'address'>]: ethers.Contract | AbacusContracts;
+};
 
-export interface IAbacusContracts<Addresses, Contracts> {
-  readonly addresses: Addresses;
-  readonly contracts: Contracts;
-  reconnect(connection: Connection): void;
-}
+export type AbacusAddresses = {
+  [key: string]: types.Address | AbacusAddresses;
+};
 
-export interface ContractsBuilder<
-  Addresses,
-  Contracts extends IAbacusContracts<Addresses, any>,
-> {
-  new (addresses: Addresses, connection: Connection): Contracts;
+export function addresses(contracts: AbacusContracts): AbacusAddresses {
```

Comment:
It might be nice to indicate the recursive nature, either by the parameter name (`contractOrObj` or something) or in a comment